File: C++/KartikeyV/Trees/diameterOfBinaryTree.cpp
int getDiameter(TreeNode* root, int& mx){
    if(root==NULL) return 0;
    int l = getDiameter(root->left, mx);
    int r = getDiameter(root->right, mx);
    mx = max(mx, l+r);
    return max(l,r)+1;
}

int solve(TreeNode* root){
    //Edges = Nodes-1
    int mx = INT_MIN;
    getDiameter(root, mx);
    return mx;
}

File: Python/vedant_bhatnagar/DevsnestProblems_THA/Phase_2/Trees/diameterofBinaryTree.py
def f(root):
    if root:
        ld,lp = f(root.left)
        rd, rp = f(root.right)
        return 1+max(ld,rd), max(lp,rp,1+ld+rd)
    return 0,0

def solve(root):
    # CODE HERE
    return f(root)[1] - 1